src/Sulu/Bundle/MediaBundle/Resources/js/views/MediaOverview/MediaOverview.js

```diff
@@ -2,104 +2,158 @@
+import {CollectionInfoStore, MediaContainer} from '../../containers/MediaContainer';
 import mediaOverviewStyles from './mediaOverview.scss';
 
 const COLLECTION_ROUTE = 'sulu_media.overview';
+const MEDIA_RESOURCE_KEY = 'media';
 const COLLECTIONS_RESOURCE_KEY = 'collections';
 
 @observer
 class MediaOverview extends React.PureComponent<ViewProps> {
-    page: observable = observable();
+    mediaPage: observable = observable();
+    collectionPage: observable = observable();
     locale: observable = observable();
-    @observable title: string;
-    @observable parentId: ?string | number;
+    @observable collectionId: ?string | number;
+    @observable mediaStore: DatagridStore;
     @observable collectionStore: DatagridStore;
-    @observable collectionId: string | number;
+    collectionInfoStore: CollectionInfoStore;
     disposer: () => void;
 
     componentWillMount() {
         const {router} = this.props;
 
-        router.bindQuery('page', this.page, '1');
+        this.mediaPage.set(1);
+
+        router.bindQuery('collectionPage', this.collectionPage, '1');
         router.bindQuery('locale', this.locale);
 
-        this.disposer = autorun(this.load);
+        this.disposer = autorun(this.createStores);
     }
 
     componentWillUnmount() {
         const {router} = this.props;
-        this.disposer();
-        router.unbindQuery('locale', this.page);
-        router.unbindQuery('page', this.locale);
+
+        router.unbindQuery('collectionPage', this.collectionPage);
+        router.unbindQuery('locale', this.locale);
+        this.mediaStore.destroy();
         this.collectionStore.destroy();
+        this.collectionInfoStore.destroy();
+        this.disposer();
     }
 
-    load = () => {
+    getCollectionId() {
         const {router} = this.props;
         const {
             attributes: {
                 id,
             },
         } = router;
 
-        if (id) {
-            this.loadCollectionInfo(id);
-        }
-
-        this.createCollectionStore(id);
-    };
-
-    loadCollectionInfo(collectionId) {
-        return ResourceRequester.get(COLLECTIONS_RESOURCE_KEY, collectionId, {
-            depth: 1,
-            locale: this.locale,
-        }).then(action((collectionInfo) => {
-            const parentCollection = collectionInfo._embedded.parent;
-            this.title = collectionInfo.title;
-            this.parentId = (parentCollection) ? parentCollection.id : undefined;
-        }));
+        return id;
     }
 
-    getTitle() {
-        if (!this.collectionId) {
-            return translate('sulu_admin.all_media');
+    createStores = () => {
+        const collectionId = this.getCollectionId();
+
+        if (collectionId !== this.collectionId || !this.collectionStore) {
+            this.setCollectionId(collectionId);
+            this.createMediaStore(collectionId, this.mediaPage, this.locale);
+            this.createCollectionStore(collectionId, this.collectionPage, this.locale);
```

Comment:
I get the following error when changing locales in the media overview:
![image](https://user-images.githubusercontent.com/405874/32481198-a15de680-c392-11e7-95dd-929031e8d3ca.png)

I think that's related with what you are doing here. DatagridStores are not meant to be destroyed and created just because the language was switched, what is happening here.

So I think I would rename this function to `updateStores`, and do not create a new store if not absolutely necessary.

Reply:
Has to do with appending the data for the masonry. Fixed.